### DISCUSSION TITLE
Forge test Error: Could not instantiate forked environment with fork url

### DISCUSSION THREAD

**Discussion by Amtaf** [Cyfrin/foundry-full-course-cu#587] [Q&A] [answered]
I'm getting the error as per the screenshot below,when running 'forge test --fork-url $SEPOLIA_RPC_URL' I have confirmed my Sepolia URL is correct . I even added the rpc endpoint on foundry.toml
![Screenshot from 2023-08-09 11-38-36](https://github.com/Cyfrin/foundry-full-course-f23/assets/9764833/9a8e81b0-039e-4d97-be3a-f26b4796f8c9)

Here's my DSC project's [repo](https://github.com/Amtaf/Algorithmic-StableCoin)

**Comment by shikhar229169** [comment, marked as answer]
Create a new rpc url and try again.

**Reply by Amtaf** [reply]
Thanks ,This worked

**Comment by theirrationalone** [comment]
you're missing your metamask private key...

Try this

```terminal
'forge test --fork-url $SEPOLIA_RPC_URL --private-key $PRIVATE_KEY

```

**Reply by theirrationalone** [reply]
Make sure to execute this command before testing.
👇
```terminal
source .env
```
and don't forget to Paste your private key in .env

like this 👇

```code
PRIVATE_KEY=YOUR_METAMASK_PRIVATE_KEY
```

**Reply by Amtaf** [reply]
Did this already

**Comment by caiosabarros** [comment]
I got this error because my block number was set in the future when I tried `vm.rollFork()`

**Reply by EngrPips** [reply]
Interesting, I never used the `vm.rollFork()` cheat code before.

**Reply by EngrPips** [reply]
Interesting, I never used the `vm.rollFork()` cheat code before.
